File: BinarySearch.py
def binary_search(arr, search_value):
  if len(arr) == 0:
    return False

  low = 0
  high = len(arr) - 1

  while low <= high:
    mid = (low + high) // 2

    if arr[mid] == search_value:
      return True

    if arr[mid] > search_value:
      high = mid - 1
    else:
      low = mid + 1

  return False

my_list = [1, 3, 5, 7, 9]
print binary_search(my_list, 3)

File: javascript/BinarySearch.js
const binarySearch = (array, search_value) => {
  if (array.length === 0) return

  var low = 0
  var high = array.length - 1
  var middle

  while (low <= high) {
    middle = Math.floor((low + high) / 2)

    if (array[middle] === search_value) return middle

    if (array[middle] > search_value) {
      high = middle - 1
    } else {
      low = middle + 1
    }
  }
}

console.log(binarySearch([21, 45, 48, 90, 101, 102, 105, 110], 105))
